tests/queries/0_stateless/02242_delete_user_race.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+# Tags: race, no-fasttest, no-parallel
+
+# Test tries to reproduce a race between threads:
+# - deletes user
+# - creates user
+# - uses it as session user
+# - apply role to the user
+#
+# https://github.com/ClickHouse/ClickHouse/issues/35714
+
+set -e
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+# shellcheck source=./replication.lib
+. "$CURDIR"/replication.lib
+
+#export PS4='\nDEBUG level:$SHLVL subshell-level: $BASH_SUBSHELL \nsource-file:${BASH_SOURCE} line#:${LINENO} function:${FUNCNAME[0]:+${FUNCNAME[0]}(): }\nstatement: '
+
+
+$CLICKHOUSE_CLIENT -nm -q "
+    DROP ROLE IF EXISTS test_role_02242;
+    CREATE ROLE test_role_02242;
+"
+
+readonly REPEAT=1000
+
+function delete_user()
+{
+    local i
+    for (( i = 0; i < REPEAT; ++i ))
+    do
+        $CLICKHOUSE_CLIENT -q "DROP USER IF EXISTS test_user_02242" ||:
+    done
+}
+
+function create_and_login_user()
+{
+    local i
+    for (( i = 0; i < REPEAT; ++i ))
+    do
+        $CLICKHOUSE_CLIENT -q "CREATE USER IF NOT EXISTS test_user_02242" ||:
+        $CLICKHOUSE_CLIENT -u "test_user_02242" -q "SELECT version()" > /dev/null ||:
+    done
+}
+
+function set_role()
+{
+    local i
+    for (( i = 0; i < REPEAT; ++i ))
```

Comment:
This is not correct usage of `clickhouse_client_loop_timeout`, especially with 0.1 timeout, since it may send INT to this function before the client will execute any query, and if you will take at the logs you will see for example that there is zero `SET ROLE` queries.